### ISSUE TITLE
keepassx-community: Text and images are cut-off

### ISSUE THREAD

**Issue by TomSmeets** [NixOS/nixpkgs#95749]
On both the latest nixpkgs-unstable and on master (08639bcd134) the keepassx-community looks very broken. This is probably something to do with Qt, but I only noticed it with this app.

**To Reproduce**
1. start keepassxc

**Expected behavior**
Readable text and images

**Screenshots**
![2020-08-18-142526_1918x1158_scrot](https://user-images.githubusercontent.com/6740669/90513022-560c2280-e15f-11ea-9152-72c6db2fa897.png)

![2020-08-18-143421_958x568_scrot](https://user-images.githubusercontent.com/6740669/90513431-f19d9300-e15f-11ea-8845-a721c5a7e7ac.png)

**Notify maintainers**
@jonafato 
@turion 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.58, NixOS, 20.09.git.aa9a4f5f185 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"nixos-20.09pre238896.16fc531784a"`
 - channels(tom): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - keepassx-community
# a list of nixos modules affected by the problem
module:
```

**Comment by JJJollyjim** [comment]
Works fine for me (master on Arch Linux)

Do you have display scaling, and if so how is it configured?

**Comment by JJJollyjim** [comment]
(I feel like I've seen similar issues in the past due to display scaling misconfigurations)

**Comment by jonafato** [comment]
I also can't reproduce (NixOS unstable, GNOME).

**Comment by TomSmeets** [comment]
Hmm strange, I am not using any display scaling. Currently I am using lightdm + awesome wm on x11.

Using a different theme does not seem to make a difference.

All dialog windows also have the same issue:
![2020-08-19-112256_1918x1158_scrot](https://user-images.githubusercontent.com/6740669/90617008-7515bd80-e20e-11ea-9b18-db698c0cdb36.png)

Resizing the window has a small effect on how the text is clipped.
Tall:
![2020-08-19-112728_1029x1131_scrot](https://user-images.githubusercontent.com/6740669/90617478-0422d580-e20f-11ea-8ffc-a96effcb3cf1.png)

Less tall:
![2020-08-19-112739_997x428_scrot](https://user-images.githubusercontent.com/6740669/90617484-05540280-e20f-11ea-8bea-4b864162406a.png)

**Comment by TomSmeets** [comment]
It looks like some qt issue. However, I cannot reproduce this issue with some other qt apps.

Dolphin emu uses Qt:
![2020-08-19-113206_1918x1158_scrot](https://user-images.githubusercontent.com/6740669/90617983-a5aa2700-e20f-11ea-8cfe-5699117164f9.png)

**Comment by TomSmeets** [comment]
So apparently it does have something to do with scaling. (However still not using any display scaling as far as i know)

Setting `QT_SCALE_FACTOR_ROUNDING_POLICY=Round` as seen in https://github.com/keepassxreboot/keepassxc/issues/5015#issuecomment-658133457 fixes my issue.

```bash
QT_SCALE_FACTOR_ROUNDING_POLICY=Round keepassxc
```
![2020-08-24-125728_958x568_scrot](https://user-images.githubusercontent.com/6740669/91037553-7c233e00-e609-11ea-8b7b-8152f4f2e1f5.png)

**Comment by turion** [comment]
That's weird. As far as I can see, nothing in nixpkgs ever sets that variable. So it's maybe an interaction with another QT app you're using?

**Comment by turion** [comment]
I guess we could wrap `keepassxc` in such a way that it always has this environment variable set on runtime. But I'd like to see someone else confirm the bug as well before applying such a change.

**Comment by turion** [comment]
(Either way, congrats on getting keepass usable again!)

**Comment by TomSmeets** [comment]
Hi @turion,

For now I just apply that hot-fix for myself by setting `environment.variables`. I agree that we should wait for others to have this same issue before merging any kind fix into nixpkgs.

These are the only QT related variables set before using a fix:
```
> env | grep '^QT'
QTWEBKIT_PLUGIN_PATH=/home/tom/.nix-profile/lib/mozilla/plugins/:/etc/profiles/per-user/tom/lib/mozilla/plugins/:/nix/var/nix/profiles/default/lib/mozilla/plugins/:/run/current-system/sw/lib/mozilla/plugins/
QT_PLUGIN_PATH=/home/tom/.nix-profile/lib/qt4/plugins:/home/tom/.nix-profile/lib/kde4/plugins:/etc/profiles/per-user/tom/lib/qt4/plugins:/etc/profiles/per-user/tom/lib/kde4/plugins:/nix/var/nix/profiles/default/lib/qt4/plugins:/nix/var/nix/profiles/default/lib/kde4/plugins:/run/current-system/sw/lib/qt4/plugins:/run/current-system/sw/lib/kde4/plugins
```
They don't seem to be very relevant

**Comment by TomSmeets** [comment]
This is the closest i got by bisecting:
good: 50c060d13dc55b673435f05119c968720c97b0c5
bad:   67af42e35ff6e233f301bdd26c774cef1c236025

So probably the commit d4c607aa715086f265f7a711c9dcb9e90a063a92 which updates qt5 from 5.12.7 to 5.14.2 broke keepassxc for me.

**Comment by TomSmeets** [comment]
Indeed reverting that commit, or just specifying Qt 5.12 for keepassxc works fine. See 7ff4990a96146df4abb45d9c6ca6504694cae39e.

**Comment by TomSmeets** [comment]
This issue appears to be fixed for me in `KeePassXC` version `2.6.1` :tada: 

I'm closing this issue.
